sw/device/lib/crypto/rsa_3072/rsa_3072_verify.c

```diff
@@ -2,17 +2,13 @@
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
 
-#include "sw/device/silicon_creator/lib/crypto/rsa_3072/rsa_3072_verify.h"
+#include "sw/device/lib/crypto/rsa_3072/rsa_3072_verify.h"
 
 #include "sw/device/lib/base/hardened.h"
 #include "sw/device/lib/base/macros.h"
 #include "sw/device/lib/base/memory.h"
-#include "sw/device/silicon_creator/lib/drivers/hmac.h"
-#include "sw/device/silicon_creator/lib/drivers/otbn.h"
-#include "sw/device/silicon_creator/lib/error.h"
-#include "sw/device/silicon_creator/lib/otbn_util.h"
-
-#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "sw/device/lib/dif/dif_hmac.h"
```

Comment:
@moidx we shouldn't be using DIFs here since this is shippable code, right...?

Reply:
I couldn't find any driver on top of the HMAC dif; should I implement one?

Reply:
Correct. This could be the start of the cryptolib hmac wrapper. We'll also have to do look at the otbn driver to make sure it is not using dif dependencies.

Reply:
The OTBN driver is using `dif_otbn`. Sounds like we need to change that? But I'm slightly confused -- I thought the DIFs were the "right" way to interface with the hardware components. Are they being deprecated, or is it only in certain circumstances that they aren't to be used?

Reply:
They difs are best suited for silicon verification as they aim to exercise all the functionality in the hardware. They are not intended to be used in production code as in those cases the configuration is more restricted and the implementation has to address other concerns such as hardening. 

This is part of the reason why we have rom/rom_ext drivers under `sw/device/silicon_creator/lib`

Reply:
Okay, in that case it sounds like I need to create both an OTBN and an HMAC driver in order to address this properly. Since this PR is already a pretty large change, is it okay to merge as-is and open an issue to create proper drivers?

Reply:
I think there's an OTBN driver at `sw/device/silicon_creator/lib/drivers/otbn.h` already. Similarly for HMAC. (But I don't really know how the SW side works: it's totally possible I'm missing something!)

Reply:
As an alternative to merging this as-is, I could also just copy the `silicon_creator` drivers as a starting point for `device/lib` drivers. That would actually make this PR smaller, as the code is originally based on the `silicon_creator` ones!

Reply:
> My understanding is that the `silicon_creator` drivers are for mask ROM/ROM_EXT only -- someone please correct me if I'm wrong there!

@jadephilipoom, yes, that's correct (the unified error space, assumptions, the amount of functionality and checks, sec_mmio, etc.).

> As an alternative to merging this as-is, I could also just copy the `silicon_creator` drivers as a starting point for `device/lib` drivers. That would actually make this PR smaller, as the code is originally based on the `silicon_creator` ones!

SGTM except I think these drivers should be under the crypto folder (similar to what we do for rom/rom_ext) since they don't have to be general purpose drivers and modifying the ones under `device/lib` would probably trigger additional work. @moidx @cfrantz @mcy